Framework/PythonInterface/plugins/algorithms/VesuvioThickness.py

```diff
@@ -0,0 +1,137 @@
+                             +"Y value in the first spectrum will be the last iteration")
+
+        self.declareProperty(ITableWorkspaceProperty("TransmissionWorkspace", "",
+                                                     direction=Direction.Output),
+                             doc="Output Workspace containing the iterative "
+                             +"approximation for Transmission.")
+
+
+    def _get_properties(self):
+        self._masses = self.getPropertyValue("Masses").split(',')
+        self._amplitudes = self.getPropertyValue("Amplitudes").split(',')
+        self._transmission_guess = self.getProperty("TransmissionGuess").value
+        self._thickness = self.getProperty("Thickness").value
+        self._number_density = self.getProperty("NumberDensity").value
+
+
+    def validateInputs(self):
+        self._get_properties()
+        issues = dict()
+
+        try:
+            self._masses = np.asarray([float(mass) for mass in self._masses])
+        except ValueError:
+            issues['Masses'] = ('Could not interpret \"%s\" ' % self._masses \
+                                    + 'as a list of numbers . Please ensure that ' \
+                                    + 'all inputs are valid')
+        try:
+            self._amplitudes = np.asarray([float(amp) for amp in self._amplitudes])
+        except ValueError:
+            issues['Amplitudes'] = ('Could not interpret \"%s\" ' % self._amplitudes \
+                                    + 'as a list of numbers . Please ensure that ' \
+                                    + 'all inputs are valid')
+
+        num_masses = len(self._masses)
+        num_amplitudes = len(self._amplitudes)
+        if num_masses != num_amplitudes:
+            issues['Masses'] = ('The number of masses: %d, ' % num_masses \
+                                + 'is not equal to the number of amplitudes: %d' % num_amplitudes)
+
+        return issues
+
+
+    def PyExec(self):
+        # Initialise output table workspaces
+        create_tbl_alg = self.createChildAlgorithm("CreateEmptyTableWorkspace")
+
+        create_tbl_alg.setProperty('OutputWorkspace', 'density_guesses')
+        create_tbl_alg.execute()
+        density_guesses_tbl_ws = create_tbl_alg.getProperty('OutputWorkspace').value
+        density_guesses_tbl_ws.addColumn("str", "Iteration")
+        density_guesses_tbl_ws.addColumn("double", "Density")
+        density_guesses_tbl_ws.setPlotType(0, 1)
+
+        create_tbl_alg.setProperty('OutputWorkspace', 'transmission_guesses')
+        create_tbl_alg.execute()
+        trans_guesses_tbl_ws = create_tbl_alg.getProperty('OutputWorkspace').value
+        trans_guesses_tbl_ws.addColumn("str","Iteration")
+        trans_guesses_tbl_ws.addColumn("double","Transmission")
+        trans_guesses_tbl_ws.setPlotType(0, 1)
+
+        # Unit conversions and scatter length calculation
+        self._thickness /= 200.0
```

Comment:
Could we add a few comments to the various conversion lines to we know what things like `200`, `1.66054e-24` mean?